### PR TITLE
[Citrix ADC] Add Integration Package with Service Data Stream

### PR DESCRIPTION
**Pull request by harnish-crest**
- Enhancement

## What does this PR do?

- Added 1 data stream (Service metrics).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/citrix_adc directory.
- Run the following command to run tests.
elastic-package test

## Related issues

- Relates #3746
- Relates #4322   
- Relates #4418
- Relates #4458 
- Relates #4473

## Screenshots
citrix_adc-overview-dashboard
new-service-dashboard

### CODE REVIEW

**Review by yug-elastic — APPROVED** [review]

File: packages/citrix_adc/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Use the Citrix ADC integration to collect metrics related to the service. Then visualize that data in Kibana, create alerts to notify you if something goes wrong, and reference logs when troubleshooting an issue.
+
+## Data streams
+
+The Citrix ADC integration collects metrics data.
+
+Metrics give you insight into the statistics of the Citrix ADC. Metrics data streams collected by the Citrix ADC integration include [service](https://developer-docs.citrix.com/projects/netscaler-nitro-api/en/12.0/statistics/basic/service/), so that the user could monitor and troubleshoot the performance of the Citrix ADC instances.
+
+Note:
+- Users can monitor and see the metrics inside the ingested documents for Citrix ADC in the logs-* index pattern from `Discover`.
```
Comment (muthu-mps) [inline packages/citrix_adc/_dev/build/docs/README.md:16]: This should be `metrics-*` and not `logs-*` index pattern.

Comment (harnish-crest) [reply packages/citrix_adc/_dev/build/docs/README.md:16]: We are using `httpjson` filebeat input. So user can see all the indices in `logs-*` format in Kibana `Discover`.

**Review by muthu-mps — APPROVED** [review]
Looks Good !

**Pull request merged**